### PR TITLE
Updates to Insecure keychain

### PR DESCRIPTION
**Pull request by mjangda**
- [x] Fix class name
- [x] Better error message for permissions errors
- [x] Move file to `.vip` dir
- [ ] Try to recover from permissions error (and/or at least provide next steps)
- [x] Verify permissions are correct for file (some Windows users reporting `Invalid permissions on access token file` error)
- [x] Support for multiple "service" values (e.g. uuid)
- [x] Switch to `homedir()`

### CODE REVIEW

**Comment by mjangda** [comment]
@joshbetz can you please take a look at the remaining pieces here?

File: src/lib/keychain/insecure.js [hunk 1]
```diff
@@ -12,15 +12,15 @@ import path from 'path';
...
+export default class Insecure implements Keychain {
 	file: string;
 
 	constructor( file: string ) {
 		// only current user has read-write access
 		const rw = 0o600;
 
 		let stat;
-		const tmpfile = os.tmpdir() + path.sep + file;
+		const tmpfile = os.homedir() + path.sep + file;
```
Comment (joshbetz) [inline src/lib/keychain/insecure.js]: Should we store it in a subdirectory like `.vip`?

**Comment by mjangda** [comment]
:+1:

File: src/lib/keychain/insecure.js (hunk 1 above)
Comment (mjangda) [reply src/lib/keychain/insecure.js]: Sounds like a good idea.

**Comment by joshbetz** [comment]
https://nodejs.org/api/fs.html#fs_file_system_flags

> On Windows, flags are translated to their equivalent ones where applicable, e.g. O_WRONLY to FILE_GENERIC_WRITE, or O_EXCL|O_CREAT to CREATE_NEW, as accepted by CreateFileW.

**Comment by joshbetz** [comment]
Would probably be good to switch to File Mode Constants instead of hard-coding the permissions to `0o600`.

https://nodejs.org/api/fs.html#fs_file_mode_constants

**Comment by joshbetz** [comment]
Nevermind, the constants don't magically work for Windows.

File: __tests__/lib/keychain.js [hunk 2]
```diff
@@ -52,6 +52,18 @@ describe( 'token tests (insecure)', () => {
...
+				const p = keychain.getPassword( 'first' );
+				expect( p ).resolves.toBe( 'password1' );
+
+				const p2 = keychain.getPassword( 'second' );
+				expect( p2 ).resolves.toBe( 'password2' );
+			} );
+		} );
+	} );
+
 	test( 'should correctly delete token', () => {
```
Comment (mjangda) [inline __tests__/lib/keychain.js:67]: We should add a test where we delete a one token out of multiples.

File: src/lib/keychain/insecure.js [hunk 3]
```diff
@@ -12,15 +12,27 @@ import path from 'path';
...
-export default class Secure implements Keychain {
+export default class Insecure implements Keychain {
 	file: string;
+	passwords: Object;
 
 	constructor( file: string ) {
 		// only current user has read-write access
-		const rw = 0o600;
+		let rw;
+		switch ( process.platform ) {
```
Comment (mjangda) [inline src/lib/keychain/insecure.js]: Would be good to have an inline comment referencing why we need this special handling.

File: src/lib/keychain/insecure.js [hunk 4]
```diff
@@ -35,33 +47,56 @@ export default class Secure implements Keychain {
...
 		} );
 	}
 
 	setPassword( service: string, password: string ): Promise<boolean> {
+		this.passwords[ service ] = password;
+
 		return new Promise( resolve => {
-			fs.writeFile( this.file, password, err => resolve( ! err ) );
+			const json = JSON.stringify( this.passwords );
+			fs.writeFile( this.file, json, err => resolve( ! err ) );
```
Comment (mjangda) [inline src/lib/keychain/insecure.js]: Would this (and `delete` below) fail silently? Should we surface the error somehow?

**Review by mjangda — COMMENTED** [review]
Noted a few minor things but looks good overall.

File: src/lib/keychain/insecure.js (hunk 3 above)
Comment (joshbetz) [reply src/lib/keychain/insecure.js]: Fortunately found a better way to use the built in constants so this special handling isn't necessary.

File: __tests__/lib/keychain.js (hunk 2 above)
Comment (joshbetz) [reply __tests__/lib/keychain.js:67]: Done in cb4a64d

**Review by joshbetz — APPROVED** [review]

**Comment by mjangda** [comment]
:ship:

**Pull request merged**